forte/data/data_pack.py

```diff
@@ -1539,38 +1542,57 @@ def get(  # type: ignore
         Args:
             entry_type: The type of entries requested.
             range_annotation: The
-                range of entries requested. If `None`, will return valid
-                entries in the range of whole data pack.
+                range of entries requested. This value can be given by an
+                entry object or the ``tid`` of that entry. If `None`, will
+                return valid entries in the range of whole data pack.
             components: The component (creator)
                 generating the entries requested. If `None`, will return valid
                 entries generated by any component.
             include_sub_type: whether to consider the sub types of
                 the provided entry type. Default `True`.
+            get_raw: boolean to indicate if the entry should be returned in
+                its primitive form as opposed to an object. False by default
 
         Yields:
             Each `Entry` found using this method.
         """
-        entry_type_: Type[EntryType] = as_entry_type(entry_type)
+        # Convert entry_type to str
+        entry_type_ = (
+            get_full_module_name(entry_type)
+            if not isinstance(entry_type, str)
+            else entry_type
+        )
+
+        # pylint: disable=protected-access
+        # Check if entry_type_ represents a valid entry
+        if not self._data_store._is_subclass(entry_type_, Entry):
+            raise ValueError(
+                f"The specified entry type [{entry_type}] "
+                f"does not correspond to a "
+                f"`forte.data.ontology.core.Entry` class"
+            )
 
         def require_annotations(entry_class=Annotation) -> bool:
-            if issubclass(entry_type_, entry_class):
+            if self._data_store._is_subclass(entry_type_, entry_class):
                 return True
-            if issubclass(entry_type_, Link):
+
+            curr_class: Type[EntryType] = as_entry_type(entry_type_)
+            if issubclass(curr_class, Link):
```

Comment:
some using `issubclass` while some using `self._data_store._is_subclass`?

Reply:
Yes that is because `curr_class` is an `EntryType` object. `self._data_store._is_subclass` requires a `str`. The reason this is done is because we don't currently have a way of getting `link.childtype, link.parenttype, or group.membertype` without converting `entry_type_`  back to class as we have discussed [here](https://github.com/asyml/forte/pull/900#discussion_r955458229)

Reply:
https://github.com/asyml/forte/issues/915